### PR TITLE
feat(components/editabletext): create component

### PR DESCRIPTION
**Pull request by ncomont**
**What is the problem this PR is trying to solve?**

EditableText only exist inside the SubHeader component.

**What is the chosen solution to this problem?**

Extract it as a new component.

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/EditableText/EditableText.component.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { translate } from 'react-i18next';
+import Skeleton from '../Skeleton';
+import { Action } from '../Actions';
+import InlineForm from './InlineForm.component';
+import theme from './EditableText.scss';
+import I18N_DOMAIN_COMPONENTS from '../constants';
+
+function EditableText({ text, editMode, loading, inProgress, onEdit, disabled, t, ...rest }) {
```
Comment (charlesnguyen) [inline packages/components/src/EditableText/EditableText.component.js]: EditableText should be used in SubHeaderBar

File: output/components.sasslint.txt [hunk 2]
```diff
@@ -4,8 +4,11 @@
 src/Drawer/Drawer.scss
   14:4  warning  Statement must begin on a new line  brace-style
 
+src/EditableText/EditableText.scss
+  38:3  error  Duplicate properties are not allowed within a block  no-duplicate-properties
```
Comment (talend-glorieux) [inline output/components.sasslint.txt]: To be fixed.

File: packages/components/src/EditableText/EditableText.component.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+						bsStyle="link"
+						className={classNames(
+							theme['tc-editable-text-text-pencil'],
+							'tc-editable-text-text-pencil',
+						)}
+						disabled={disabled}
+						hideLabel
+					/>
+				</div>
+			)}
```
Comment (talend-glorieux) [inline packages/components/src/EditableText/EditableText.component.js]: Not a big fan of that giant ternary. But I can live with it :smile:

File: packages/components/src/EditableText/EditableText.component.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+	t: PropTypes.func,
+};
+
+EditableText.defaultProps = {
+	editMode: false,
+	loading: false,
+	inProgress: false,
+};
+
+export default translate(I18N_DOMAIN_COMPONENTS)(EditableText);
```
[the remark this answers is not in the document]
Comment (ncomont) [reply packages/components/src/EditableText/EditableText.component.js]: ✅

File: packages/components/src/EditableText/EditableText.component.js (hunk 1 above)
Comment (ncomont) [reply packages/components/src/EditableText/EditableText.component.js]: ✅

File: output/components.sasslint.txt (hunk 2 above)
Comment (ncomont) [reply output/components.sasslint.txt]: ✅

File: packages/components/src/EditableText/EditableText.component.js (hunk 3 above)
Comment (ncomont) [reply packages/components/src/EditableText/EditableText.component.js]: ✅

File: packages/components/src/SubHeaderBar/InputTitleSubHeader/InputTitleSubHeader.component.js [hunk 5]
```diff
@@ -1,50 +0,0 @@
...
-			{iconId && (
-				<Icon
-					name={iconId}
-					className={classNames(theme['tc-subheader-details-icon'], 'tc-subheader-details-icon')}
-				/>
-			)}
-			{editMode ? (
-				<InlineFormSubHeader title={title} {...rest} />
-			) : (
-				<TitleSubHeader title={title} disabled={inProgress} {...rest} />
```
Comment (charlesnguyen) [inline packages/components/src/SubHeaderBar/InputTitleSubHeader/InputTitleSubHeader.component.js]: image
You should consider also this case

Comment (ncomont) [reply packages/components/src/SubHeaderBar/InputTitleSubHeader/InputTitleSubHeader.component.js]: ✅

**Pull request merged**